zokrates_codegen/src/lib.rs

```diff
@@ -2237,11 +2237,36 @@ impl<'ast, T: Field> Flattener<'ast, T> {
                     .cloned()
                     .map(|p| self.layout.get(&p.id.id).cloned().unwrap().into())
                     .collect();
+
                 let outputs: Vec<Variable> = assignees
                     .into_iter()
                     .map(|assignee| self.use_variable(&assignee))
                     .collect();
-                let directive = FlatDirective::new(outputs, Solver::Zir(function), inputs);
+
+                let mut substitution_map = HashMap::default();
+                for (index, p) in function.arguments.iter().enumerate() {
+                    let new_id = format!("i{}", index).into();
+                    substitution_map.insert(p.id.id.clone(), new_id);
```

Comment:
This substitution is done so that we can deduplicate function calls later. We want functions with the same logic to hash to the same thing eg. `(a, b) -> a + b` is logically the same as `(c, d) -> c + d`. Both functions are transformed to `(i0, i1) -> i0 + i1`